### PR TITLE
Deprecate AFS (Andrew File System) support in Netatalk 3

### PR DESCRIPTION
**Pull request by dgsga**
Cherry-picked from my macOS fork

### CODE REVIEW

**Comment by rdmark** [comment]
@dgsga There is some minor additional cleanup of AFS code for you to consider:

- doc/manual/configuration.xml -- talks about AFS Kerberos
- dead ifdef macros in:
- etc/afpd/unix.h 
- etc/uams/uams_guest.c
- etc/uams/uams_passwd.c
- etc/uams/uams_pgp.c
- the AFPVOL_AFSGVS flag is unused: include/atalk/volume.h

Please let me know if there's a reason to keep these. :)

**Comment by dgsga** [comment]
> @dgsga There is some minor additional cleanup of AFS code for you to consider:
> 
> * doc/manual/configuration.xml -- talks about AFS Kerberos
> * dead ifdef macros in:
> * etc/afpd/unix.h
> * etc/uams/uams_guest.c
> * etc/uams/uams_passwd.c
> * etc/uams/uams_pgp.c
> * the AFPVOL_AFSGVS flag is unused: include/atalk/volume.h
> 
> Please let me know if there's a reason to keep these. :)

Well spotted. Done.

File: doc/manual/configuration.xml [hunk 1]
```diff
@@ -710,27 +710,6 @@ basedir regex = /usr/home</programlisting></para>
...
-
-                <secondary>"AFS Kerberos" UAM (Kerberos IV)</secondary>
-              </indexterm> UAMs (suitable to use <ulink
-            url="http://web.mit.edu/macdev/KfM/Common/Documentation/faq.html">Kerberos
-            v4 based authentication</ulink> and AFS file servers)</para>
-
-            <para>Use <filename>"--enable-krb4-uam"</filename> at compile time
-            to activate the build of this UAM.</para>
-          </listitem>
-
```
Comment (rdmark) [inline doc/manual/configuration.xml:733]: Isn't removing all the Kerberos docs overkill? We still support the stock UAMs.

Comment (dgsga) [reply doc/manual/configuration.xml:733]: Have just removed Kerberos IV UAM reference as --enable-krb4-uam is not an option for Netatalk 3 (only Kerberos V support here, --enable-krbV-uam)

Comment (rdmark) [reply doc/manual/configuration.xml:733]: Aha, I had forgotten about that, after using netatalk2 too much. :)

**Pull request merged**